[release/3.1] Clean out Ubuntu 16.04 testing

dougbu opened the pull request:
- a variant of #32894
  - use Ubuntu 18.04 agents for testing in Docker containers on Helix
- remove use of old Centos and Debian queues
- move to newer OSX.1014.Amd64.Open queue and Fedora.33.Amd64.Open Docker image
- add a couple of queues we test against in 'main'

dougbu committed:
[release/3.1] Clean out Ubuntu 16.04 testing
- a variant of #32894
  - use Ubuntu 18.04 agents for testing in Docker containers on Helix
- remove use of old Centos and Debian queues
- move to newer OSX.1014.Amd64.Open queue and Fedora.33.Amd64.Open Docker image
- add a couple of queues we test against in 'main'

dougbu commented on eng/targets/Helix.Common.props:
```diff
@@ -18,23 +18,22 @@
     <HelixAvailableTargetQueue Include="Windows.81.Amd64.Open" Platform="Windows" />
     <HelixAvailableTargetQueue Include="Windows.7.Amd64.Open" Platform="Windows" />
     <HelixAvailableTargetQueue Include="Windows.10.Amd64.EnterpriseRS3.ASPNET.Open" Platform="Windows" EnableByDefault="false" />
```
We test on this plus Windows.10.Amd64.ClientRS4.VS2017.Open. Should I upgrade to more recent Windows.10 builds❔

HaoK replied:
Seems like we should try to leave the matrix alone on older releases shouldn't we?  Otherwise we might just create extra test failures/differences to investigate across branches?  Like maybe the idea would be we only upgrade when required, like with 16.04 being EOL

dougbu replied:
@HaoK overall, are you suggesting I leave the Windows queues alone and do _not_ add OSX 11.00 or Ubuntu 20.04 but clean up or upgrade from CentOS 7, Debian 8, Fedora 28, OSX 10.13, and Ubuntu 16.04❔

(I would pick Ubuntu 18.04 over 20.4 at the moment because it's more commonly used. EOL is 2 years in the future.)

HaoK replied:
Correct, I'd basically only update 16.04 -> 18.04 and just drop Centos7/Debian8/Fedora28.  If we don't have to update OSX 1013 -> 1014, or any windows versions, I wouldn't do either.

Given that we aren't working actively in this branch, minimizing churn with the queues seems better than being proactive to update things (assuming things are mostly working on the current queues).

Tratcher replied:
What's the oldest Win10 version officially supported by dotnet? We should at least test the oldest and newest.

dougbu replied:
> What's the oldest Win10 version officially supported by dotnet?

@Pilchie @halter73 is there a Windows 10 or OSX support matrix for our servicing releases somewhere❔

dougbu commented on eng/targets/Helix.Common.props:
```diff
@@ -18,23 +18,22 @@
-    <HelixAvailableTargetQueue Include="OSX.1013.Amd64.Open" Platform="OSX" />
-    <HelixAvailableTargetQueue Include="Ubuntu.1604.Amd64.Open" Platform="Linux" />
+    <HelixAvailableTargetQueue Include="OSX.1014.Amd64.Open" Platform="OSX" />
+    <HelixAvailableTargetQueue Include="OSX.1100.Amd64.Open" Platform="OSX" />
```
Is it fine / worthwhile to add the OSX.1100.Amd64.Open queue❔

HaoK replied:
Doesn't seem worth adding more coverage in an old release especially since we don't generally see many issues due to OSX differences (basically we get more noise for no reward).  But if you are upgrading, maybe just pick one? 1100?

dougbu replied:
Removed

dougbu commented on eng/targets/Helix.Common.props:
```diff
@@ -18,23 +18,22 @@
     <HelixAvailableTargetQueue Include="Ubuntu.1804.Amd64.Open" Platform="Linux" />
-    <HelixAvailableTargetQueue Include="Centos.7.Amd64.Open" Platform="Linux" />
-    <HelixAvailableTargetQueue Include="Debian.8.Amd64.Open" Platform="Linux" />
+    <HelixAvailableTargetQueue Include="Ubuntu.2004.Amd64.Open" Platform="Linux" />
```
Is it fine / worthwhile to add Ubuntu.2024.Amd64.Open queue❔

HaoK replied:
Doesn't seem worth if 1804 isn't going away anytime soon

dougbu replied:
Removed

dougbu commented on eng/targets/Helix.Common.props:
```diff
@@ -18,23 +18,22 @@
+    <HelixAvailableTargetQueue Include="OSX.1100.Amd64.Open" Platform="OSX" />
     <HelixAvailableTargetQueue Include="Ubuntu.1804.Amd64.Open" Platform="Linux" />
-    <HelixAvailableTargetQueue Include="Centos.7.Amd64.Open" Platform="Linux" />
-    <HelixAvailableTargetQueue Include="Debian.8.Amd64.Open" Platform="Linux" />
```
Am I correct we no longer support Centos 7 or Debian 8 (or Fedora 28) even for 3.1.x❔

HaoK replied:
I think its safe to remove these regardless, have we've ever found a product issue from Centos/Debian helix testing?  So I don't think they add much value in an old servicing branch

dougbu commented on eng/targets/Helix.Common.props:
```diff
@@ -18,23 +18,22 @@
   <ItemGroup Condition="'$(IsWindowsOnlyTest)' != 'true' AND '$(TargetArchitecture)' == 'arm64'">
     <!-- arm64 queues -->
-    <HelixAvailableTargetQueue Include="(Debian.9.Arm64.Open)Ubuntu.1604.Arm64.Docker.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-9-helix-arm64v8-a12566d-20190807161036" Platform="Linux" />
+    <HelixAvailableTargetQueue Include="Ubuntu.1804.ArmArch.Open" Platform="Linux" />
```
Why don't we test on bare Ubuntu 18.04 ARM64 in 'main'❔

HaoK replied:
I don't remember how we settled of what queues we picked for arm64 testing, so seems fine to update/clean things up for main.  But I'd still lean towards less changes over conformity in terms of queues for release/3.1 vs main

dougbu replied:
Removed from here

dougbu replied:
#33054

dougbu commented on eng/targets/Helix.Common.props:
```diff
@@ -18,23 +18,22 @@
-    <HelixAvailableTargetQueue Include="(Alpine.38.Arm64)Ubuntu.1604.Arm64.Docker@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.8-helix-arm64v8-46e69dd-20190327215724" Platform="Linux" />
-    <HelixAvailableTargetQueue Include="(Ubuntu-1804.Arm64.Open)Ubuntu.1604.Arm64.Docker.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-6f28fa9-20190606004102" Platform="Linux" />
+    <HelixAvailableTargetQueue Include="(Alpine.38.Arm64)Ubuntu.1804.ArmArch@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.8-helix-arm64v8-46e69dd-20190327215724" Platform="Linux" />
     -->
```
This commented-out bit doesn't seem to make much sense anymore. Alpine 3.8 is very old and we don't test w/ Alpine ARM64 images in 'main'. Remove❔

HaoK replied:
Yeah remove

dougbu replied:
Done

dougbu reviewed (commented):
Lots of questions&hellip;

dougbu committed:
Remove queue additions
- don't add OSX 11.00 or Ubuntu 20.04 queues
- leave Ubuntu 16.04 removal
  - also use Ubuntu 18.04 base for Debian 9 ARM64 testing
- leave OSX 10.13 -> 10.14 and Fedora 28 -> 33 upgrades

nit: remove old commented-out mention of Alpine.38.Arm64

dougbu commented:
Hmm, @HaoK and @dotnet/aspnet-build this change may be completely irrelevant. I can't find a pipeline associated w/ our helix-test.yml file and ci.yml doesn't mention Helix at all. I am no longer sure we test using Helix anywhere w/ the release/3.1 branch. Did we remove it at some point or perhaps never finish adding it❔

HaoK commented:
Right, that's what I was sorta asking a few weeks ago, I am pretty sure 5.0 was the first time we actually turned helix on for real, 3.1 was as manual/optional thing I think

dougbu commented:
Not relevant and would need @HaoK's #33034 to get through CI
